File: src/main/java/org/radarbase/appserver/exception/FileStorageException.java
package org.radarbase.appserver.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
public class FileStorageException extends RuntimeException {
    private static final long serialVersionUID = -793674245766939L;

    public FileStorageException(String message) {
        super(message);
    }

    public FileStorageException(String message, Object object) {
        super(message + " " + object.toString());
    }
}

File: appserver-legacy/src/main/kotlin/org/radarbase/appserver/exception/FileStorageException.kt
package org.radarbase.appserver.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus
import java.io.Serial

@Suppress("unused")
@ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
class FileStorageException : RuntimeException {
    companion object {
        @Serial
        private const val serialVersionUID: Long = -793674245766939L
    }

    constructor(message: String) : super(message)

    constructor(message: String, `object`: Any) : super("$message $`object`")
}
